subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1382,4 +1431,11 @@ public boolean isConfigurationTransitive() {
         }
     }
 
+    private class AllArtifactsProvider implements PublishArtifactSetProvider {
```

Comment:
Instead of this I'd lazily create the `outgoing` field and call getAllArtifacts at that point.